stat.go

```diff
@@ -125,6 +125,32 @@ func ChiSquare(obs, exp []float64) float64 {
 	return result
 }
 
+// CircularMean returns the circular mean of the dataset.
+// If weights is nil then all of the weights are 1. If weights is not nil, then
+// len(x) must equal len(weights).
+func CircularMean(x, weights []float64) float64 {
+	if weights != nil && len(x) != len(weights) {
+		panic("stat: slice length mismatch")
+	}
+
+	var aX, aY float64
+	for i := range x {
+		if weights != nil {
+			aX += weights[i] * math.Cos(x[i])
+			aY += weights[i] * math.Sin(x[i])
+			continue
+		}
+		aX += math.Cos(x[i])
+		aY += math.Sin(x[i])
+	}
+
+	res := math.Atan2(aY, aX)
+	if math.Abs(res) < 1e-14 {
```

Comment:
I'm not sure what @kortschak had in mind, but this shouldn't be based on a tolerance. Maybe

````
res := math.Atan2(aY, aX)
// correct for negative 0
if res == 0 { 
      return 0
}
````